compiler/passes/checkResolved.cpp

```diff
@@ -543,28 +545,147 @@ checkCalls()
 }
 
 
+static bool isExternType(Type* t) {
+  // narrow references are OK but not wide references
+  if (t->isWideRef())
+    return false;
+
+  ClassTypeDecorator d = CLASS_TYPE_UNMANAGED_NONNIL;
+  // unmanaged or borrowed classes are OK
+  if (isClassLikeOrManaged(t) || isClassLikeOrPtr(t))
+    d = removeNilableFromDecorator(classTypeDecorator(t));
+
+  TypeSymbol* ts = t->symbol;
+
+  EnumType* et = toEnumType(t);
+
+  return t->isRef() ||
+         d == CLASS_TYPE_BORROWED ||
+         d == CLASS_TYPE_UNMANAGED ||
+         (et && et->isConcrete()) ||
+         (ts->hasFlag(FLAG_TUPLE) && ts->hasFlag(FLAG_STAR_TUPLE)) ||
+         ts->hasFlag(FLAG_GLOBAL_TYPE_SYMBOL) ||
+         ts->hasFlag(FLAG_DATA_CLASS) ||
+         ts->hasFlag(FLAG_C_PTR_CLASS) ||
+         ts->hasFlag(FLAG_C_ARRAY) ||
+         ts->hasFlag(FLAG_EXTERN) ||
+         ts->hasFlag(FLAG_EXPORT); // these don't exist yet
+}
+
+static bool isExportableType(Type* t) {
+
+  if (t == dtString || t == dtBytes) {
+    // string/bytes are OK in export functions
+    // because they are converted to wrapper
+    // functions
+    return true;
+  }
+
+  return isExternType(t);
+}
+
+static void externExportTypeError(FnSymbol* fn, Type* t) {
```

Comment:
@e-kayrakli and I had a discussion about this - I'll leave the c_string error for string only (because while that error might make sense for bytes, we probably want to get rid of c_string anyway).

I'll add a comment about how an `export` routine with a `string` argument or return type is already translated into a wrapper by this point in compilation. (In that event, this check helps us to be sure that we don't have cases that are falling through that process in some way).